### PR TITLE
History ui

### PR DESCRIPTION
**Pull request by kyracq**
https://censusviz.appspot.com/

Notes:
- Query History doesn't show up immediately after sign in, because doing this led to getHistory being called twice and duplicates in the carousel
- When on first page of slider/carousel, the back button should take you to the last page (when on last page forward button should take you to first page)
- We don't have timestamps in history elements yet, so for now I just reverse the history array which seems to give approximately the desired result

### CODE REVIEW

File: src/main/webapp/style.css [hunk 1]
```diff
@@ -307,3 +307,46 @@ input[type="color"] {
   padding: 8px;
   height: 18px;
 }
+
```
Comment (shulijones) [inline src/main/webapp/style.css:310]: We may want to start dividing this document up into sections using big comments as headers; we've got a lot of different groups of css going on here now.

Comment (kyracq) [reply src/main/webapp/style.css:331]: Done!

File: src/main/webapp/login-status.js [hunk 2]
```diff
@@ -53,13 +41,13 @@ function loginInit() {
...
-      gapi.signin2.render('sign-in-btn', {
-        'onsuccess': onSignIn,
-      });
+      gapi.signin2.render('sign-in-btn');
       displayLoginStatus();
+      toggleHistory(false);
     }).then(() => {
-      if (getLoginStatus()) {
-        getHistory(getUserId());
+      const isLoggedIn = getLoginStatus();
```
[the remark this answers is not in the document]
Comment (kyracq) [reply src/main/webapp/login-status.js]: Fixed

File: src/main/webapp/query-control.js [hunk 3]
```diff
@@ -289,6 +330,13 @@ function replaceValueIfEmpty(dataListId) {
   }
 }
 
+// Change the year of data being visualized
+function changeYear(yearParam) {
+  const year = yearParam.value;
+  document.getElementById('year-list').value = year;
```
[the remark this answers is not in the document]
Comment (kyracq) [reply src/main/webapp/query-control.js]: @shulijones

[the remark this answers is not in the document]
Comment (shulijones) [reply src/main/webapp/query-control.js]: I can change that! Thanks.

File: src/main/webapp/query-control.js [hunk 4]
```diff
@@ -76,47 +76,86 @@ function putHistory(personType, action, location, year, userId) {
...
+function createSlide(title, url, location) {
+  const historyItem = document.createElement('li');
+  historyItem.className = 'splide__slide';
+  const link = document.createElement('a');
+  link.href = url;
+  const titleNode = document.createTextNode(title);
+  const imgWrapper = document.createElement('div');
+  imgWrapper.className = 'img-wrapper';
+  const image = document.createElement('img');
+  image.alt = 'Small U.S. map';
```
Comment (shulijones) [inline src/main/webapp/query-control.js:88]: now that the map is different depending on location, you could append ' of ' + stateInfo[location].name for county queries

File: src/main/webapp/style.css [hunk 5]
```diff
@@ -118,34 +132,40 @@ input[type=submit]:hover {
...
+#query-form {
+  display: inline-flex;
 }
 
+#query {
+  text-align: center;
+}
+
+/* Visualzations */
```
Comment (shulijones) [inline src/main/webapp/style.css:168]: nit: typo here

**Comment by kyracq** [comment]
> LGTM! I love the static maps for each state. One nitpick is that when you sign out and then sign in again without the page reloading, the history only shows up after you've made a query. Because history disappears instantaneously it might be good to also have it appear again instantaneously.

Yeah, I also agree that'd be ideal but for some reason doing that caused duplicates in the carousel

**Pull request merged**